numba/targets/randomimpl.py

```diff
@@ -327,18 +335,41 @@ def _randrange_impl(context, builder, start, stop, step, state):
     nbits = builder.trunc(builder.call(fn, [n, cgutils.true_bit]), int32_t)
     nbits = builder.sub(ir.Constant(int32_t, ty.width), nbits)
 
-    bbwhile = builder.append_basic_block("while")
-    bbend = builder.append_basic_block("while.end")
-    builder.branch(bbwhile)
-
-    builder.position_at_end(bbwhile)
-    r = get_next_int(context, builder, state_ptr, nbits)
-    r = builder.trunc(r, ty)
-    too_large = builder.icmp_signed('>=', r, n)
-    builder.cbranch(too_large, bbwhile, bbend)
+    if state == "np":
+        # Counting leading number of zeros above results in one too many bits
```

Comment:
What is it about the implementation in NumPy that forces this slightly contrived behaviour? It'd be useful to add some comments here about the reasons behind it.